src/main/java/cz/smarteon/loxone/user/UserGroup.java

```diff
@@ -0,0 +1,123 @@
+package cz.smarteon.loxone.user;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import cz.smarteon.loxone.LoxoneUuid;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Getter
+public class UserGroup {
+
+    /**
+     * Enum class containing constants for usergroup type
+     */
+    @RequiredArgsConstructor
+    public enum UserGroupType {
+        NORMAL(0),
+        @Deprecated
+        DEPRECATED_ADMIN(1),
+        ALL(2),
+        NONE(3),
+        ADMIN(4);
+
+        @Getter
+        private final int value;
+    }
+
+    /**
+     * Enum class containing constants for usergroup rights
+     */
+    @RequiredArgsConstructor
+    public enum UserGroupRights {
+        NONE(0),
```

Comment:
I wonder whether these rights are not some bitmap of flags, see
* admin as `11111111111111111111111111111111`
* none as all zeros
* etc.

No need for change, just thinking out loud